src/containers/home.js

```diff
@@ -5,20 +5,45 @@ import { connect } from 'react-redux'
 import { fetch, ConnectedPostList } from './ConnectedPostList'
 import PostEditor from '../components/PostEditor'
 import CoverImagePage from '../components/CoverImagePage'
-import { setCurrentCommunityId } from '../actions'
+import { navigate, fetchCommunity } from '../actions'
+import { setCurrentCommunityIdLocalAndRemote } from '../actions/util'
+import { communityUrl } from '../routes'
+const { object } = React.PropTypes
 
-const makeComponent = (subject, title, showEditor) => props => {
-  const { location: { query }, currentUser: { id } } = props
+const makeComponent = (subject, title, showEditor) => React.createClass({
+  propTypes: {
+    location: object,
+    currentUser: object
+  },
 
-  return <CoverImagePage>
-    {showEditor && <PostEditor/>}
-    <ConnectedPostList {...{subject, id, query}}/>
-  </CoverImagePage>
-}
+  componentDidMount: function () {
+    const { query: { rd }, pathname } = this.props.location
+    rd && window.history.replaceState({}, 'Hylo', pathname)
+  },
+
+  render: function () {
+    const { location: { query }, currentUser: { id } } = this.props
+
+    return <CoverImagePage>
+      {showEditor && <PostEditor/>}
+      <ConnectedPostList {...{subject, id, query}}/>
+    </CoverImagePage>
+  }
+})
 
 const wrapComponent = (subject, ...args) => compose(
-  prefetch(({ dispatch, params, currentUser: { id }, query }) => {
-    dispatch(setCurrentCommunityId('all'))
+  prefetch(({ dispatch, params, currentUser: { id, settings: { currentCommunityId } }, query, path }) => {
+    if (query.rd && currentCommunityId) {
+      if (currentCommunityId !== 'all') {
+        // we only have the id, so have to fetch to get the slug
```

Comment:
also, if you want to test this, just pull the prefetch function out and export it so that you can call it from a test. then just give it mock arguments and test that it dispatches `navigate` with the right url